src/macros.rs

```diff
@@ -2,6 +2,31 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+/// Wrap all libcryptsetup_rs_sys calls in the macro. It will expand to a
+/// feature-flagged mutex lock call. If the `mutex` feature is not enabled, it is
+/// a no-op.
+macro_rules! mutex {
+    ( $libcryptsetup_call:expr ) => {{
+        #[cfg(feature = "mutex")]
+        #[allow(unused_variables)]
+        let lock = match $crate::MUTEX.lock() {
+            Ok(l) => Some(l),
+            Err(e) => {
+                log::warn!(
+                    "The synchronization mutex can not longer be locked! \
+                    Locking failed with error: {}; see Rust's documentation \
+                    on Mutex poisoning here to decide how to proceed: \
+                    https://doc.rust-lang.org/std/sync/struct.Mutex.html#errors",
+                    e,
+                );
+                None
+            }
+        };
+
+        $libcryptsetup_call
```

Comment:
I believe that every invocation that requires the ```mutex!``` must also be necessarily ```unsafe```. Is there any benefit to including the ```unsafe``` in this macro definition?

Reply:
Let me think about this. I can see a benefit in terms of code readability, but I'm still thinking through if it'll be confusing obscuring that from view. I know in other places, we've avoided returning errors in macros and have opted instead to put the `?` operator outside of the macro. Do you think the same thing applies here?

Reply:
I'm not sure...one thing I thought was that if the unsafe call around the libcryptsetup invocation was mandatory (if missing a compiler error results), including the unsafe in the macro would ensure that the mutex was always added in the right place in new code.

Reply:
That seems like a good reason to me. I'm willing to update the PR with that.

Reply:
Resolved.